### PR TITLE
Fix the rewire() of rrt* path planning (#550) 

### LINKED ISSUE #550: RRT* with reeds-sheep seems to produce illegal paths
**Describe the bug**
Sometimes the path produced by RRT* with reeds-sheep script rrt_star_reeds_shepp.py seems not allowed: there are sharp corners in the path.

**Expected behavior**
allowed path

**Screenshots**
Screenshot from 2021-10-02 20-34-39

Screenshot from 2021-10-02 20-41-58

**Desktop (please complete the following information):**
 - Python version 3.8.10

### PR DESCRIPTION
**Pull request by RyderCRD**
#### Reference issue

fix #550 

#### What does this implement/fix?

rrt_star_reeds_shepp.py needs its own version of rewire(), which not only updates path_x , path_y, but also path_yaw when rewiring. 
Before this pull request, it uses the rewire() in rrt_star.py, which only cares about path_x, and path_y, thus issue #550 might occur.

The range of phi in rrt_star_reeds_shepp.py is changed from (0, 0.99\*math.pi) to (0.01\*math.pi, 0.99*math.pi) for the sake of speed.

#### Additional information

Two tests are added into tests/test_rrt_star_reeds_shepp.py.

#### CheckList
- [x] Did you add an unittest for your new example or defect fix?
- [x] Did you add documents for your new example?
- [x] All CIs are green? (You can check it after submitting)

### CODE REVIEW

File: PathPlanning/ReedsSheppPath/reeds_shepp_path_planning.py [hunk 1]
```diff
@@ -53,17 +54,14 @@ def mod2pi(x):
 
 def straight_left_straight(x, y, phi):
     phi = mod2pi(phi)
-    if y > 0.0 and 0.0 < phi < math.pi * 0.99:
+    # only take phi in (0.01*math.pi, 0.99*math.pi) for the sake of speed.
+    # phi in (0, 0.01*math.pi) will make test2() in test_rrt_star_reeds_shepp.py
+    # extremely time-consuming, since the value of xd, t will be very large.
+    if math.pi * 0.01 < phi < math.pi * 0.99 and np.sign(y) != 0:
```
Comment (AtsushiSakai) [inline PathPlanning/ReedsSheppPath/reeds_shepp_path_planning.py:60]: np.sign is needed? just y != 0 is not fine?

File: PathPlanning/RRTStar/rrt_star.py [hunk 2]
```diff
@@ -225,13 +226,12 @@ def rewire(self, new_node, near_inds):
...
 
             if no_collision and improved_cost:
-                near_node.x = edge_node.x
-                near_node.y = edge_node.y
-                near_node.cost = edge_node.cost
-                near_node.path_x = edge_node.path_x
-                near_node.path_y = edge_node.path_y
-                near_node.parent = edge_node.parent
-                self.propagate_cost_to_leaves(new_node)
+                for node in self.node_list:
```
Comment (AtsushiSakai) [inline PathPlanning/RRTStar/rrt_star.py:229]: Why should we search parent here? 
The previous implementation was wrong?

**Review by RyderCRD — COMMENTED** [review]
Reviewed.

**Review by AtsushiSakai — APPROVED** [review]

**Comment by AtsushiSakai** [comment]
@RyderCRD LGTM. Thank you!!

**Pull request merged**